Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean

```diff
@@ -203,6 +203,15 @@ theorem Basis.parallelepiped_reindex (b : Basis ι ℝ E) (e : ι ≃ ι') :
     (congr_arg _root_.parallelepiped (b.coe_reindex e)).trans (parallelepiped_comp_equiv b e.symm)
 #align basis.parallelepiped_reindex Basis.parallelepiped_reindex
 
+theorem Basis.mem_parallelepiped [DecidableEq ι] {b : Basis ι ℝ E} {a : E} :
+    a ∈ (b.parallelepiped : Set E) ↔ ∀ i, b.repr a i ∈ Set.Icc 0 1 := by
+  simp_rw [coe_parallelepiped, mem_parallelepiped_iff, Set.mem_Icc, Pi.le_def, ← forall_and]
+  refine ⟨?_, fun h => ⟨b.repr a, fun i => h i, (b.sum_repr a).symm⟩⟩
+  rintro ⟨_, h, rfl⟩ i
+  simpa only [LinearEquiv.map_sum, SMulHomClass.map_smul, repr_self, Finsupp.smul_single,
+    smul_eq_mul, mul_one, Finset.sum_apply', Finsupp.single_apply, Finset.sum_ite_eq',
+    Finset.mem_univ, ite_true] using h i
```

Comment:
I have a more-bundled version of this result [here](https://github.com/leanprover-community/mathlib/pull/19013/files#diff-10dd661e90f9d04ff1ba015bde0648b9e60bec561ebeca5f2576e9e8b8a04e4eR74-R91), which it would be nice to have as well; does the bundling help at all?

Reply:
Yes, your version seems better.